File: 1122Programming/leetcode/231. Power of Two/Solution.py
'''

https://leetcode.com/problems/two-sum/
Problem ID: 1. Two Sum

Author: GU_7

'''
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False
        while n != 1:
            if n & 1:
                return False
            n >>= 1
        return True

File: 1122Programming/leetcode/231. Power of Two/Solution.swift
/*
 *
 * https://leetcode.com/problems/power-of-two/
 * Problem ID: 231. Power of Two
 * 
 * Author: Ateto
 * 
 */

class Solution {
    func isPowerOfTwo(_ n: Int) -> Bool {
        if n <= 0 {
            return false
        }
        var tmp = n
        while tmp > 1 {
            if tmp&1 == 1 {
                return false
            }
            tmp >>= 1
        }
        return true
    }
}